pip/utils/__init__.py

```diff
@@ -54,6 +55,37 @@ def get_prog():
     return 'pip'
 
 
+def default_user_site():
+    # Avoid circular import, the running_under_virtualenv should probably be
+    # in pip.utils anyways TBH.
+    from pip.locations import running_under_virtualenv, distutils_scheme
+
+    # If we're running inside of a virtual environment, we do not want to
+    # install to the --user directory.
+    if running_under_virtualenv():
+        return False
+
+    # If the Python we're running under does not have their user packages
+    # enabled then we do not want to install to the user directory since it
+    # may or may not work or exist.
+    if not site.ENABLE_USER_SITE:
+        return False
+
+    # If any of our potentional locations for writing files is not writable by
+    # us then we want to use the --user scheme instead of the --global scheme.
+    # TODO: We should figure out a way to make this work for --root and
+    #       --isolated and such options as well.
+    # TODO: Figure out if this works when the directories don't exist.
+    for path in distutils_scheme("").values():
+        if not os.access(path, os.W_OK):
```

Comment:
Retrying with ``--user`` seems like the wrong answer to me to be honest. It sounds like it'd be easy to lead to half way installed things and it would be a lot more fragile. If ``os.access`` doesn't work we can probably do something based on trying to open file with ``w`` to test it.